lib/elixir/src/elixir_rewrite.erl

```diff
@@ -184,15 +185,19 @@ rewrite(?access, _DotMeta, 'get', Meta, [Arg, _], Env)
   elixir_errors:compile_error(Meta, ?m(Env, file),
     "the Access syntax and calls to Access.get/2 are not available for the value: ~ts",
     ['Elixir.Macro':to_string(Arg)]);
-rewrite(?string_chars, _DotMeta, 'to_string', _Meta, [String], _File) when is_binary(String) ->
+rewrite(?list_chars, _DotMeta, 'to_char_list', _Meta, [List], _Env) when is_list(List) ->
```

Comment:
Alright, I thought, for a start we could just go same way as we do in `List.Chars` for lists: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/list/chars.ex#L36.
Shall we restrict there as well?